### PR TITLE
Feature/niad 376 ammendment gp to ha address

### PR DESCRIPTION
**Pull request by Alicja-Kainos**

### CODE REVIEW

File: src/intTest/resources/outbound_uat_data/amendment/app-j-4.edifact.dat [hunk 1]
```diff
@@ -0,0 +1,14 @@
+UNB+UNOA:2+TES5+XX11+200610:1438+00000001'
```
Comment (mattd-kainos) [inline src/intTest/resources/outbound_uat_data/amendment/app-j-4.edifact.dat]: The "app-j" tests are taken directly from "Appendix J" of the spec. This one has been modified to test - I think only amending the house and street name. Could you change the filename maybe `house-street-names-only.edifact..`?

File: src/intTest/resources/outbound_uat_data/amendment/live-1.fhir.ignore.json [hunk 2]
```diff
@@ -29,6 +29,11 @@
       "op": "replace",
       "path": "/address/0/line/4",
       "value": "KENT"
+    },
```
Comment (mattd-kainos) [inline src/intTest/resources/outbound_uat_data/amendment/live-1.fhir.ignore.json:32]: live-1 only contains addresses. Can this also be enabled?

**Review by grzegorz-robak — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java [hunk 3]
```diff
@@ -31,11 +33,25 @@ public String getFormattedSimpleValue() {
         return value.get();
     }
 
+    public String getNullSafeFormattedSimpleValue() {
+        if (op == AmendmentPatchOperation.REMOVE) {
```
Comment (mattd-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java:36]: A REMOVE operation should never have a "value" property. Why is this needed?

Comment (mattd-kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java:36]: Oh I see its doing the mapping to %

File: src/intTest/resources/outbound_uat_data/amendment/live-1.fhir.ignore.json (hunk 2 above)
Comment (Alicja-Kainos) [reply src/intTest/resources/outbound_uat_data/amendment/live-1.fhir.ignore.json:32]: live-1 contains also haCipher, that cannot be proceed right now. I get error that: "Couldn't decode recipient: KC01"

File: src/intTest/resources/outbound_uat_data/amendment/app-j-4.edifact.dat (hunk 1 above)
Comment (Alicja-Kainos) [reply src/intTest/resources/outbound_uat_data/amendment/app-j-4.edifact.dat]: ok, renamed

**Review by grzegorz-robak — DISMISSED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java [hunk 4]
```diff
@@ -8,6 +8,8 @@
 import lombok.Setter;
 import lombok.ToString;
 
+import org.apache.logging.log4j.util.Strings;
```
Comment (mattd-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java]: I suspect this is not the Strings class you want

Comment (Alicja-Kainos) [reply src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java]: no, I will just add a constant instead :)

**Review by grzegorz-robak — APPROVED** [review]

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonAddress.java [hunk 5]
```diff
@@ -3,12 +3,15 @@
...
 import org.apache.commons.lang3.StringUtils;
+
+import lombok.EqualsAndHashCode;
 import uk.nhs.digital.nhsconnect.nhais.model.edifact.message.EdifactValidationException;
 
 import java.util.Objects;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+@EqualsAndHashCode(callSuper = false)
```
Comment (mattd-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonAddress.java:14]: Someone finally got tired of all the compiler warnings 😆

File: src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java [hunk 6]
```diff
@@ -17,6 +17,7 @@
 public class AmendmentPatch {
 
     private static final String REMOVE_INDICATOR = "%";
+    private static final String EMPTY_STRING = "";
```
Comment (mattd-kainos) [inline src/main/java/uk/nhs/digital/nhsconnect/nhais/model/jsonpatch/AmendmentPatch.java:20]: StringUtils.EMPTY from commons-lang3 is what we're using in other parts of the application

**Review by mattd-kainos — APPROVED** [review]
Approved with optional comment about using StringUtils.EMPTY
